draw.it.client/src/pages/gameplay/GameplayScreen.jsx

```diff
@@ -74,13 +80,19 @@ export default function GameplayScreen() {
         return () => {
             gameplayConnection.off("ReceiveMessage");
             gameplayConnection.off("ReceiveTurnStarted");
+            gameplayConnection.off("ReceiveWordToDraw");
             gameplayConnection.off("ReceiveRoundStarted");
             gameplayConnection.off("ReceiveRoundEnded");
             gameplayConnection.off("ReceiveGameEnded");
             gameplayConnection.off("ReceivePlayerStatuses");
         };
     }, [gameplayConnection, roomId]);
-    
+
+    useEffect(() => {
+        if (!myName || playerStatuses.length === 0) return;
+        const drawer = playerStatuses.some(p => p.name === myName && p.isDrawer);
+        setIsDrawer(drawer);
+    }, [myName, playerStatuses]);
```

Comment:
Please revert this change to the old implementation:
```suggestion
    const isDrawer = playerStatuses.some(
        (player) => player.isDrawer && player.name === myName 
    );
```

`useEffect` shouldn't be used here, as `isDrawer` is updated during rerender (on the state change of `playerStatuses` or `myName`). I'm referring to the React docs:
> If you’re not trying to synchronize with some external system, you probably don’t need an Effect.

Also, you can read more [here](https://react.dev/learn/you-might-not-need-an-effect)